src/StructuredLogViewer.Core/NodeQueryMatcher.cs

```diff
@@ -332,17 +353,38 @@ public SearchResult IsMatch(BaseNode node)
                     else
                     {
                         string fullText = field;
-                        if (node is NameValueNode named && fullText == named.Name)
+                        if (node is NameValueNode named && fullText == named.Name && AdditionalKeywords.Count == 0)
                         {
                             // if we matched a property in the name, show value as well since it's useful
                             fullText = named.ToString();
                         }
-
-                        result.AddMatch(fullText, word);
+                        // NameValueNode is special case have to check in witch field to search
+                        if (node is NameValueNode && (searchInName ||  searchInValue))
+                        {
+                            if (searchInName && j == 1 && word == AdditionalKeywords["name"])

```

Comment:
I think this will be expensive at runtime. Using a dictionary is very flexible but adds a lot of cost. Let's just hard code name and value strings.

Reply:
I've cleaned up and added two strings and two boolean instead of dictionary, it's seems to be better.. . Thanks for your remarks :-)